frame/im-online/src/mock.rs

```diff
@@ -161,13 +197,37 @@ impl pallet_authorship::Trait for Runtime {
 
 impl Trait for Runtime {
 	type AuthorityId = UintAuthorityId;
+	type OffchainAuthorityId = ImOnlineAuthorityId;
 	type Event = ();
-	type Call = Call;
-	type SubmitTransaction = SubmitTransaction;
 	type ReportUnresponsiveness = OffenceHandler;
 	type SessionDuration = Period;
 }
 
+impl<LocalCall> frame_system::offchain::SendTransactionTypes<LocalCall> for Runtime where
+	Call: From<LocalCall>,
+{
+	type OverarchingCall = Call;
+	type Extrinsic = Extrinsic;
+}
+
+impl frame_system::offchain::SigningTypes for Runtime {
+	type Public = PublicWrapper;
+	type Signature = Signature;
+}
+
+impl<LocalCall> frame_system::offchain::CreateSignedTransaction<LocalCall> for Runtime where
+	Call: From<LocalCall>,
+{
+	fn create_transaction<C: frame_system::offchain::AppCrypto<Self::Public, Self::Signature>>(
+		call: Call,
+		_public: PublicWrapper,
+		_account: u64,
+		nonce: u64,
+	) -> Option<(Call, <Extrinsic as ExtrinsicT>::SignaturePayload)> {
+		Some((call, (nonce, ())))
+	}
+}
+
```

Comment:
I am a bit confused why we need to specify things like signature when a module needs to just send unsigned transactions? the precious model was good in this aspect.

Reply:
After looking at the stacking pallet, i agree that it shouldn't be required. I'll see what i can do about it.

Reply:
Implemented `SubmitTransaction` without needing to specify any crypto related stuff (which is the case for the staking pallet)